File: arena-jface/src/main/java/org/uqbar/lacar/ui/impl/jface/radiogroup/JFaceRadioGroupBuilder.java
package org.uqbar.lacar.ui.impl.jface.radiogroup;

import org.eclipse.swt.widgets.Composite;
import org.uqbar.lacar.ui.impl.jface.JFaceContainer;
import org.uqbar.lacar.ui.impl.jface.lists.JFaceAbstractListBuilder;

/**
 * Jface builder for radiogroup selectors
 * 
 * @author jfernandes
 */
public class JFaceRadioGroupBuilder<T> extends JFaceAbstractListBuilder<T, RadioGroupViewer, RadioGroup> {

	public JFaceRadioGroupBuilder(JFaceContainer container) {
		super(container);
	}

	@Override
	protected RadioGroupViewer createViewer(Composite jFaceComposite) {
		RadioGroupViewer theViewer = new RadioGroupViewer(jFaceComposite);
		this.initialize(theViewer.getRadioGroup());
		return theViewer;
	}

}

File: arena-jface/src/main/scala/org/uqbar/lacar/ui/impl/jface/builder/lists/JFaceRadioGroupBuilder.scala
package org.uqbar.lacar.ui.impl.jface.builder.lists

import org.eclipse.swt.widgets.Composite
import org.uqbar.lacar.ui.impl.jface.builder.traits.JFaceContainer
import org.uqbar.lacar.ui.impl.jface.radiogroup.RadioGroup
import org.uqbar.lacar.ui.impl.jface.radiogroup.RadioGroupViewer

class JFaceRadioGroupBuilder[T](container: JFaceContainer) extends JFaceAbstractListBuilder[T, RadioGroupViewer, RadioGroup](container) {

  override def createViewer(jFaceComposite: Composite) = {
    val theViewer = new RadioGroupViewer(jFaceComposite)
    initialize(theViewer.getControl)
    theViewer
  }

}